sidm/tools/sidm_processor.py

```diff
@@ -162,18 +162,14 @@ def process(self, events):
 
         return {events.metadata["dataset"]: out}
 
-    def make_vector(self, objs, collection, type_id=None, mass=None, charge=None):
+            # use nanoevents.Muon behaviors for dsa muons
+    def make_vector(self, objs, collection, fields,  type_id=None, mass=None, charge=None, dxy=None, ):
         shape = ak.ones_like(objs[collection].pt)
-        return vector.zip(
-            {
-                "part_type": objs[collection]["type"] if type_id is None else type_id*shape,
-                "charge": objs[collection].charge if charge is None else charge*shape,
-                "pt": objs[collection].pt,
-                "eta": objs[collection].eta,
-                "phi": objs[collection].phi,
-                "mass": objs[collection].mass if mass is None else mass*shape,
-            }
-        )
+        forms = {f : objs[collection][f] if f in objs[collection].fields else 0*shape for f in fields}
```

Comment:
do I understand correctly that this will assign a value of 0 to any field that doesn't exist for a particular object? I remember this being necessary to fix the issues related to `ak.concatonate`, but I worry that it might lead to confusion down the road. One way to avoid possible confusion would be to remove these fake fields from the objects after line 195.